.github/workflows/build.yml

```diff
@@ -79,28 +117,83 @@ jobs:
+    needs:
+      - calculate-tags
+      - push-ghcr
+    steps:
+      - uses: sigstore/cosign-installer@v3.7.0
+
+      - name: Install skopeo & podman
+        run: |
+          sudo apt update
+          sudo apt install -y skopeo podman
+
+      # Workaround bug where capital letters in your GitHub username make it impossible to push to GHCR.
+      # https://github.com/macbre/push-to-ghcr/issues/12
+      - name: Lowercase Registry
+        id: registry_case
+        uses: ASzc/change-string-case-action@v6
+        with:
+          string: ${{ env.IMAGE_REGISTRY }}
+
+      # Login with docker for cosign
       - name: Login to GitHub Container Registry
         uses: docker/login-action@v3
-        if: github.event_name != 'pull_request'
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      # Sign container
-      - uses: sigstore/cosign-installer@v3.7.0
-        if: github.event_name != 'pull_request'
+      # Login with podman for buildah
+      - name: Log in to ghcr.io
+        uses: redhat-actions/podman-login@4934294ad0449894bcd1e9f191899d7292469603 # v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Combine manifests
+        id: combine-manifests
+        run: |
+          alias_tags=(${{ needs.calculate-tags.outputs.alias_tags }})
+          sha_short=${{ needs.calculate-tags.outputs.sha_short }}
+          date=${{ needs.calculate-tags.outputs.date }}
+
+          tags=("${alias_tags[@]}" "$sha_short")
+
+          # If date is not empty string, append the date to the tags
+          if [[ -n "$date" ]]; then
+            tags+=("$date")
+          fi
+
+          digest=""
+
+          for tag in "${tags[@]}"; do
+            buildah manifest create ${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}:${tag} \
+              ${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}:${tag}-aarch64 \
+              ${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}:${tag}-x86_64
+
+            buildah manifest push ${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}:${tag} docker://${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}:${tag}
+
+            digest=$(skopeo inspect docker://${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}:${tag} --format '{{.Digest}}')
+          done
+
+          echo "digest: $digest"
+
+          echo "digest=$digest" >> $GITHUB_OUTPUT
 
       - name: Sign container image
-        if: github.event_name != 'pull_request'
         run: |
-          cosign sign -y --key env://COSIGN_PRIVATE_KEY ${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}@${TAGS}
+          cosign sign --recursive -y --key env://COSIGN_PRIVATE_KEY ${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}@${TAGS}
```

Comment:
recursive flag will take the top level manifest, sign it, then find all referenced manifests and sign those too. Tested in my repo and this worked.

<img width="700" alt="Screenshot 2025-02-09 at 22 30 05" src="https://github.com/user-attachments/assets/697baaba-c429-41bc-abff-b0d1342ff09e" />